app/frontend/src/features/communities/discussion/CreateDiscussionForm.tsx

```diff
@@ -0,0 +1,123 @@
+import { Card, Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import TextField from "components/TextField";
+import { CANCEL } from "features/constants";
+import { useForm } from "react-hook-form";
+import makeStyles from "utils/makeStyles";
+
+import {
+  CREATE_NEW_DISCUSSION_TITLE,
+  CREATE_NEW_DISCUSSION_TOPIC,
+  NEW_DISCUSSION_TITLE,
+  NEW_DISCUSSION_TOPIC,
+  POST,
+} from "../constants";
+import { CreateDiscussionInput, useNewDiscussionMutation } from "../hooks";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "& > :not(:last-child)": {
+      marginBlockEnd: theme.spacing(3),
+    },
+    marginBlockEnd: theme.spacing(5),
+    padding: theme.spacing(3),
+  },
+  title: {
+    marginTop: 0,
+  },
+  form: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(3),
+    },
+  },
+  actionButtonsContainer: {
+    "& > * + *": {
+      marginInlineStart: theme.spacing(3),
+    },
+    display: "flex",
+    justifyContent: "flex-end",
+  },
+}));
+
+export interface CreateDiscussionFormProps {
+  communityId: number;
+  onCancel?(): void;
+  onPost?(): void;
+}
+
+type CreateDiscussionData = Omit<CreateDiscussionInput, "ownerCommunityId">;
+
+export default function CreateDiscussionForm({
+  communityId,
+  onCancel,
+  onPost,
+}: CreateDiscussionFormProps) {
+  const classes = useStyles();
+  const {
+    handleSubmit,
+    reset: resetForm,
+    register,
+  } = useForm<CreateDiscussionData>({
+    mode: "onBlur",
+  });
+
+  const {
+    error,
+    mutate: createDiscussion,
+    reset: resetMutation,
+  } = useNewDiscussionMutation(handleSuccess);
+
+  function handleSuccess() {
+    resetForm();
+    resetMutation();
+  }
+
+  const handleCancel = () => {
+    onCancel?.();
+    resetForm();
+    resetMutation();
+  };
+
+  const handlePost = () => {
+    onPost?.();
+  };
+
+  const onSubmit = handleSubmit((data) => {
+    createDiscussion({ ...data, ownerCommunityId: communityId });
+  });
+
+  return (
+    <Card className={classes.root}>
+      <Typography className={classes.title} variant="h2">
+        {CREATE_NEW_DISCUSSION_TITLE}
+      </Typography>
+      {error && <Alert severity="error">{error.message}</Alert>}
+      <form className={classes.form} onSubmit={onSubmit}>
+        <TextField
+          fullWidth
+          id="title"
+          name="title"
+          inputRef={register({ required: true })}
+          label={NEW_DISCUSSION_TITLE}
+        />
+        <TextField
+          fullWidth
+          id="content"
+          name="content"
+          inputRef={register({ required: true })}
+          label={NEW_DISCUSSION_TOPIC}
+          multiline
+          placeholder={CREATE_NEW_DISCUSSION_TOPIC}
+          rows={6}
+        />
+        <div className={classes.actionButtonsContainer}>
+          <Button onClick={handlePost} type="submit">
+            {POST}
+          </Button>
```

Comment:
Good point. I wonder if it's better to collapse the form in `onSuccess` instead, since even calling in `onSubmit` closes the form too quickly for us to see the loading state

Reply:
Yep. Actually, if this is just being used for closing, maybe the prop should just be `onClose` like MUI

Reply:
I think I'm going to rename it as `onPostSuccess` and call it in `onSuccess` in the hook, since it just happens that successful posting closes the form. I think it's not unreasonable to expect we _might_ have to something extra given design keeps on tweaking this area still... so probably doesn't hurt to keep it semi-generic.